### PR TITLE
maestro azure postgres entra auth

### PR DESCRIPTION
**Pull request by geoberle**
### What this PR does

* leverage entra auth for postgres access
* bump maestro to support entra auth
* includes various maestro fixes mentioned in https://github.com/Azure/ARO-HCP/pull/970

Jira: https://issues.redhat.com/browse/ARO-8831
Link to demo recording:

### CODE REVIEW

File: maestro/server/Makefile [hunk 1]
```diff
@@ -21,5 +22,7 @@ deploy:
 		--set image.base=${IMAGE_BASE} \
 		--set image.tag=${IMAGE_TAG} \
 		--set database.containerizedDb=${USE_CONTAINERIZED_DB} \
-		--set database.ssl='${USE_DATABASE_SSL}'
+		--set database.ssl='${USE_DATABASE_SSL}' \
+		$(if $(filter false,$(USE_CONTAINERIZED_DB)),--set database.host=$${DATABASE_HOST}) \
```
Comment (janboll) [inline maestro/server/Makefile:26]: Is this some fallback for other environments?

File: dev-infrastructure/modules/postgres/postgres-access.bicep [hunk 2]
```diff
@@ -34,10 +34,16 @@ var sqlScriptLines = [
...
   'GRANT ALL PRIVILEGES ON DATABASE ${databaseName} TO "${newUserName}";'
-  'GRANT SELECT, INSERT, UPDATE, DELETE ON ALL TABLES IN SCHEMA public TO "${newUserName}";'
+  'GRANT ALL ON SCHEMA public TO "${newUserName}";'
+  'GRANT USAGE ON SCHEMA public TO "${newUserName}";'
+  'GRANT ALL PRIVILEGES ON ALL TABLES IN SCHEMA public TO "${newUserName}";'
+  '\\c ${databaseName};'
+  'GRANT ALL ON SCHEMA public TO "${newUserName}";'
+  'GRANT USAGE ON SCHEMA public TO "${newUserName}";'
+  'GRANT ALL PRIVILEGES ON ALL TABLES IN SCHEMA public TO "${newUserName}";'
 ]
```
[the remark this answers is not in the document]
Comment (geoberle) [reply dev-infrastructure/modules/postgres/postgres-access.bicep:44]: no that was me trying to figure out what permissions are required for PG 15+ auth to work. i can most likely clean up the stuff before `\c DB` but i need to test if that breaks something for CS which is still on PG 12 right now

[the remark this answers is not in the document]
Comment (geoberle) [reply dev-infrastructure/modules/postgres/postgres-access.bicep:44]: * removing the parts before `\c DB` breaks things for CS as long as CS keeps running on postgres < 15
* adding the parts after `\c DB` is required for maestro as it runs on postgres >= 15

File: maestro/server/Makefile (hunk 1 above)
Comment (geoberle) [reply maestro/server/Makefile]: i cleaned this up a bit, but yes the db host is either a static string for container DBs or an azure DB host

**Review by janboll — APPROVED** [review]

**Pull request merged**